src/FileWriterTask.cpp

```diff
@@ -25,119 +25,120 @@ json hdf_parse(std::string const &Structure) {
-    std::string ServiceID_,
+    std::string const &ServiceID_,
     std::shared_ptr<KafkaW::ProducerTopic> StatusProducer_)
-    : ServiceID(ServiceID_), StatusProducer(StatusProducer_) {
+    : ServiceId(ServiceID_), StatusProducer(std::move(StatusProducer_)) {
   using namespace std::chrono;
-  _id = static_cast<uint64_t>(
+  Id = static_cast<uint64_t>(
       duration_cast<nanoseconds>(system_clock::now().time_since_epoch())
           .count());
-  _id = (_id & uint64_t(-1) << 16) | (n_FileWriterTask_created & 0xffff);
+  Id = (Id & uint64_t(-1) << 16) | (n_FileWriterTask_created & 0xffff);
   ++n_FileWriterTask_created;
 }
 
 FileWriterTask::~FileWriterTask() {
   LOG(Sev::Debug, "~FileWriterTask");
-  _demuxers.clear();
+  Demuxers.clear();
   try {
-    hdf_file.close();
+    hdfFile.close();
     if (StatusProducer) {
-      logEvent(StatusProducer, StatusCode::Close, ServiceID, JobID,
+      logEvent(StatusProducer, StatusCode::Close, ServiceId, JobId,
                "File closed");
     }
   } catch (std::exception const &E) {
     if (StatusProducer) {
-      logEvent(StatusProducer, StatusCode::Fail, ServiceID, JobID,
+      logEvent(StatusProducer, StatusCode::Fail, ServiceId, JobId,
                fmt::format("Exception while finishing FileWriterTask: {}",
                            E.what()));
     }
   }
 }
 
-FileWriterTask &FileWriterTask::set_hdf_filename(std::string hdf_output_prefix,
-                                                 std::string hdf_filename) {
-  this->hdf_output_prefix = hdf_output_prefix;
-  this->hdf_filename = hdf_filename;
-  return *this;
+void FileWriterTask::setFilename(std::string const &Prefix,
+                                 std::string const &Name) {
+  if (Prefix.empty()) {
+    Filename = Name;
+  } else {
+    Filename = Prefix + "/" + Name;
+  }
 }
 
-void FileWriterTask::add_source(Source &&source) {
-  if (UseHDFSWMR) {
-    source.HDFFileForSWMR = &hdf_file;
+void FileWriterTask::addSource(Source &&Source) {
+  if (swmrEnabled()) {
+    Source.HDFFileForSWMR = &hdfFile;
   }
-  bool found = false;
-  for (auto &d : _demuxers) {
-    if (d.topic() == source.topic()) {
-      d.add_source(std::move(source));
-      found = true;
+
+  // If source already exists then replace
+  for (auto &d : Demuxers) {
+    if (d.topic() == Source.topic()) {
+      d.add_source(std::move(Source));
+      return;
     }
   }
-  if (!found) {
-    _demuxers.emplace_back(source.topic());
-    auto &d = _demuxers.back();
-    d.add_source(std::move(source));
-  }
-}
 
-void FileWriterTask::hdf_init(std::string const &NexusStructure,
-                              std::string const &ConfigFile,
-                              std::vector<StreamHDFInfo> &stream_hdf_info) {
-  filename_full = hdf_filename;
-  if (!hdf_output_prefix.empty()) {
-    filename_full = hdf_output_prefix + "/" + filename_full;
-  }
+  // Add new source
+  Demuxers.emplace_back(Source.topic());
+  auto &d = Demuxers.back();
```

Comment:
I generally dislike single character variables.
```suggestion
  auto &LastDemuxer = Demuxers.back();
```